### ISSUE TITLE
update to version 3.4 throw ex :[Right-hand side of 'instanceof' is not an object]

### ISSUE THREAD

**Issue by liva0126** [1904labs/dom-to-image-more#184]

**Comment by liva0126** [comment]
当我回退到版本3.3时，此问题就修复了。

**Comment by liva0126** [comment]
This issue was fixed when I retreated to version 3.3.

**Comment by IDisposable** [comment]
Can you give me a repro on JSFiddle? I can't really debug from here.

**Comment by pschmiedelenbw** [comment]
Same error here, unfortunately I also cannot give you a repro. The call is something like
```javascript
domtoimage.toPng(someElement, { bgColor: '#fff', height, width }).then(dataUrl => ...)
```
It seems the problem is in `isShadowRoot()` which when called with a `document` element calls `getWindow()` which tries to access the `ownerDocument`, however that is not set for `document` elements.

**Comment by IDisposable** [comment]
Pushing v3.4.1 to see if that fixes it... reverting part of a PR I merged.

**Comment by IDisposable** [comment]
Please verify or deny if v3.4.2 fixes this @pschmiedelenbw and @liva0126

**Comment by pschmiedelenbw** [comment]
That seems to fix the problem, thanks!

**Comment by cWenyu** [comment]
Hi @IDisposable , it seems breaks the slotted element, can you please double check release 3.4.2 with https://github.com/1904labs/dom-to-image-more/issues/178

**Comment by IDisposable** [comment]
Fixed in v3.4.3

**Comment by ktdd** [comment]
v3.4.3 is broken here with this error, 3.4.2 works. No ShadowDom in use, just a regular element.

**Comment by cWenyu** [comment]
I have no issues to use 3.4.3 with regular elements (no shadow dom), see the example: https://jsfiddle.net/CWenyu/rh4Ljqa5/5/

**Comment by cannot-hero** [comment]
I use 3.4.3, il still has this probem, so I retreated to version 3.3

**Comment by codesculpture** [comment]
@IDisposable 
The library is broken, in this [commit (specifically this line)](https://github.com/1904labs/dom-to-image-more/commit/0f612a3fdbdc9301a30086e9b56aa8e00fd3ebec#diff-270a29e4e36906febc65fe59a2b7035f5a4f079af8ae57680ee0707332268051R668)

`Object.hasOwn(value, 'getRootNode')` which is not producing same output as `'getRootNode' in value`

I am not sure the reason behind this but it is producing different output.
Kindly look at this , since we were depending on "disableEmbedOptions" which is introduced in the same commit

**Comment by codesculpture** [comment]
> I use 3.4.3, il still has this probem, so I retreated to version 3.3

3.4.2 consist of revert of this issue, so its safe to use 3.4.2 in case you want `disableEmbedFonts` options.
@cannot-hero

**Comment by IDisposable** [comment]
How about v3.4.4 just released @codesculpture @cannot-hero @ktdd

**Comment by cannot-hero** [comment]
@IDisposable  i just tried v3.4.4,  still have this problem
![image](https://github.com/user-attachments/assets/356eb9eb-6425-4e2f-a2a8-d4563bbf2923)

**Comment by IDisposable** [comment]
I can't reproduce this anywhere, can you contribute a test to the .spec.js file?

**Comment by codesculpture** [comment]
Can we reopen this (this issue is not listed on my home page, where it seems only opened issues would pop up there, i need to manually come to the repo find this issue to comment)
@IDisposable

**Comment by codesculpture** [comment]
Duplicated this issue (since it is closed) [here](https://github.com/1904labs/dom-to-image-more/issues/189)

**Comment by codesculpture** [comment]
This been addressed here https://github.com/1904labs/dom-to-image-more/issues/189

**Comment by IDisposable** [comment]
MUCH better fix Released in [v3.8.0](https://www.npmjs.com/package/dom-to-image-more/v/3.8.0)

**Comment by IDisposable** [comment]
MUCH better fix Released in [v3.8.0](https://www.npmjs.com/package/dom-to-image-more/v/3.8.0)
